packages/messageport/lib/messageport_tizen.dart

```diff
@@ -0,0 +1,164 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+const MethodChannel _channel = MethodChannel('tizen/messageport');
+
+/// Signature for a callback receiving message on messageport.
+///
+/// This is used by [LocalPort.register].
+typedef OnMessageReceived = Function(dynamic message, [RemotePort? remotePort]);
+
+/// Local port to receive messages
+class LocalPort {
+  LocalPort._(this._id, this._portName, this._isTrusted);
+
+  /// Register messageport and sets listener
+  ///
+  /// Names of already registered ports has to be unique.
+  /// Exception will be thrown when using on already registered port
+  void register(OnMessageReceived onMessage) {
+    if (_registeredPorts.contains(portName)) {
+      throw Exception('Port $portName is already registered');
+    }
+
+    _eventChannel = EventChannel('tizen/messageport$_id');
+    _streamSubscription =
+        _eventChannel.receiveBroadcastStream().listen((dynamic event) {
+      if (event is Map) {
+        final Map<dynamic, dynamic> map = event;
+        final dynamic message = map['message'];
+        if (map.containsKey('remotePort')) {
+          final String remoteAppId = map['remoteAppId'] as String;
+          final String remotePort = map['remotePort'] as String;
+          final bool isTrusted = map['isTrusted'] as bool;
+          onMessage(message, RemotePort._(remoteAppId, remotePort, isTrusted));
+        } else {
+          onMessage(message);
+        }
+      }
+    });
+
+    _registeredPorts.add(portName);
+  }
+
+  /// Unregisters messageport
+  Future<void> unregister() async {
+    await _streamSubscription?.cancel();
+    _registeredPorts.remove(portName);
+  }
+
+  /// Checks whether local port is trusted
+  bool get trusted {
+    return _isTrusted;
+  }
+
+  /// Returns port name
+  String get portName {
+    return _portName;
+  }
+
+  /// Checks whether local port is registered
+  bool get registered {
+    return _registeredPorts.contains(_portName);
+  }
+
+  final int _id;
+  final bool _isTrusted;
+  final String _portName;
+  StreamSubscription<dynamic>? _streamSubscription;
+  late EventChannel _eventChannel;
+
+  static final Set<String> _registeredPorts = <String>{};
```

Comment:
I have changed the implementation and deleted `_registeredPorts` set. All values sent from dart are validated by native side, so nothing wrong should happen in case of invalid parameters - exception will be thrown.
Also there is nothing wrong with sharing the same port name by two or more applications - ports are defined by its name and Application Id.